buildSrc/src/test/java/org/elasticsearch/gradle/precommit/DependencyLicensesTasksIT.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.gradle.precommit;
+
+import org.elasticsearch.gradle.test.GradleIntegrationTestCase;
+import org.gradle.testkit.runner.GradleRunner;
+
+public class DependencyLicensesTasksIT extends GradleIntegrationTestCase {
```

Comment:
Do you think there's any added benefit of the integration test over the unit test ?
The only one I see is that   it checks that the task is wired correctly, but it does a lot more checks that seem redundant. The unit test is probably sufficient in this case. A unit test for `PrecommitTasks` could check the wiring if we wanted to ( not saying it needs to go in this PR ).